MapboxNavigation/RouteMapViewController.swift

```diff
@@ -315,9 +317,15 @@ extension RouteMapViewController: NavigationMapViewDelegate {
     func navigationMapView(_ mapView: MGLMapView, imageFor annotation: MGLAnnotation) -> MGLAnnotationImage? {
         return delegate?.navigationMapView(mapView, imageFor:annotation)
     }
+    
+    func mapViewDidFinishLoadingMap(_ mapView: MGLMapView) {
+        hasFinishedLoadingStyle = true
+    }
 
     @objc(navigationMapView:shouldUpdateTo:)
     func navigationMapView(_ mapView: NavigationMapView, shouldUpdateTo location: CLLocation) -> CLLocation? {
+        
+        guard hasFinishedLoadingStyle else { return nil }
```

Comment:
So you’re actually right about needing to track `hasFinishedLoadingStyle` separately. `style` is `nil` until the style finishes loading, but sources such as Streets may take a little longer to load. But instead of `guard`, we need to modify the existing giant `if` statement.